src/util/d.ts

```diff
@@ -29,15 +28,15 @@ function d(tagNameOrFactory: TagNameOrFactory, options: DOptions = {}, children?
 			}
 		};
 	}
-	else if (typeof tagNameOrFactory === 'function') {
+
+	if (typeof tagNameOrFactory === 'function') {
 		return {
 			factory: tagNameOrFactory,
 			options: <WidgetOptions<WidgetState>> options
 		};
 	}
-	else {
-		throw new Error('Unsupported tagName or factory type');
-	}
+
+	throw new Error('Unsupported tagName or factory type');
```

Comment:
Again, thinking out loud... as we found out with Maquette, a throw during a render will totally halt any renders in forever at the moment.  Would an error in `d()` actually be terminal.  Also speaks to `dojo-debug` and how we instrument our code (cc/ @rishson)